core/src/client/ErrorHandler/ErrorHandler.tsx

```diff
@@ -18,38 +18,51 @@ interface ErrorHandlerState {
     showDetails: boolean;
 }
 
-export class ErrorHandler extends React.PureComponent<ErrorHandlerProps, ErrorHandlerState> {
-    constructor(props) {
-        super(props);
+export class ErrorHandler extends PureComponent<ErrorHandlerProps, ErrorHandlerState> {
 
-        this.state = {
-            showDetails: false,
-        };
-    }
+    state: Readonly<ErrorHandlerState> = { showDetails: false };
 
     onBackClick = (): void => {
-        window.history.back();
+        // Back button - takes you back to the previous page if available
+        // and to the ‘home’ folder if not possible to go back to the previous page
+        if (window.history.length !== 1) {
+            // browsers like chrome stores their homepage as first item
+            window.history.back();
+        } else {
+            window.location.href = ActionURL.getBaseURL(false);
+        }
     };
 
     onViewDetailsClick = (): void => {
-        this.setState(() => ({
-            showDetails: true,
-        }));
+        const { showDetails } = this.state;
+        this.setState({ showDetails: !showDetails });
     };
 
     render() {
-        const { errorType, message } = this.props.context;
+        const { errorDetails } = this.props.context;
         const { showDetails } = this.state;
 
         return (
             <>
-                <ErrorTopSection
-                    errorType={errorType}
-                    onBackClick={this.onBackClick}
-                    onViewDetailsClick={this.onViewDetailsClick}
-                />
-                {/* TODO : ErrorPage, following section in next story*/}
-                {showDetails && <h3 className="labkey-error">{message}</h3>}
+                <div className="error-details-body">
+                    <div className="row">
+                        <div className="col-md-8">
+                            <div className="labkey-error-top">
+                                {getErrorHeading()}
+                                {getSubHeading(errorDetails)}
+                                {getInstruction(errorDetails)}
+                                <button className="btn btn-info error-backButton" onClick={this.onBackClick}>
```

Comment:
I know the mockups had this as the bootstrap blue "info" button, but we should stick to our standard button classes and let our core scss (which will hopefully remove some of the bootstrap overrides in the future).
```suggestion
                                <button className="btn btn-primary error-backButton" onClick={this.onBackClick}>
```